frontend/src/app/modules/global_search/global-search-input.component.ts

```diff
@@ -199,54 +121,59 @@ export class GlobalSearchInputComponent implements OnDestroy {
     event.stopPropagation();
     event.preventDefault();
 
+    // handle click on search button
     if (ContainHelpers.insideOrSelf(this.btn.nativeElement, event.target)) {
       this.submitNonEmptySearch();
     }
   }
 
-  public redirectToWp(id:string) {
-    window.location = this.PathHelperService.workPackagePath(id) as unknown as Location;
-  }
+  // load selected item
+  public onChange($event:any) {
+    // update embeeded table and title when new search is submitted
+    this.globalSearchService.searchTerm = this.searchTerm;
 
-  private unregister() {
-    if (this.unregisterGlobalListener) {
-      this.unregisterGlobalListener();
-      this.unregisterGlobalListener = undefined;
+    let selectedOption = $event;
+    if (selectedOption.id) {  // item is a work package element
+      this.redirectToWp(selectedOption.id);
+    } else {                  // item is a 'scope' element
+      this.searchInScope(selectedOption.projectScope);
     }
   }
 
-  public submitNonEmptySearch(forcePageLoad:boolean = false) {
-    this.globalSearchService.searchTerm = this.searchValue;
-    if (this.searchValue !== '') {
-      // Work package results can update without page reload.
-      if (!forcePageLoad &&
-          this.globalSearchService.isAfterSearch() &&
-          this.globalSearchService.currentTab === 'work_packages') {
-        window.history
-          .replaceState({},
-            `${I18n.t('global_search.search')}: ${this.searchValue}`,
-            this.globalSearchService.searchPath());
+  // close menu when input field is empty
+  public toggleMenu(searchedTerm:string) {
+    (searchedTerm.trim().length > 0) ? this.ngSelectComponent.isOpen = true : this.ngSelectComponent.isOpen = false;
+  }
 
-        return;
-      }
+  public onFocus() {
+    this.expanded = true;
+    this.ngSelectComponent.filterValue = this.searchTerm;
+  }
 
-      this.globalSearchService.submitSearch();
-    }
+  public onFocusOut() {
+    this.expanded = false;
+    this.ngSelectComponent.isOpen = false;
   }
 
-  private get searchValue() {
-    return this.input.nativeElement.value;
+  // get work packages result list and append it to suggestions
+  private getSearchResult(term:string) {
+    this.autocompleteWorkPackages(term).then((values) => {
+      this.results = this.suggestions.concat(values.map((wp:any) => {
+        return { id: wp.id, subject: wp.subject, status: wp.status.name, statusId: wp.status.idFromLink, $href: wp.$href };
```

Comment:
Please use a line per hash attribute if there are that many attributes
```suggestion
        return {
          id: wp.id,
          subject: wp.subject,
          status: wp.status.name,
          statusId: wp.status.idFromLink,
          $href: wp.$href
        };
```